Engine.UnitTests/TestPlanTest.cs

```diff
@@ -638,8 +637,46 @@ public void RepeatChildStepsFailure()
             var run = plan.Execute(new IResultListener[] { new SlowResultListener() });
             Assert.AreEqual(Verdict.NotSet, run.Verdict);
         }
+        /// <summary>
+        /// Tests that a parallel step where the SuggestedNextStep is set to the current ID will repeat the step. 
+        /// </summary>
+        [Test]
+        public void ParallelSuggestedNextStepTest()
+        {
+            TestTraceListener trace = new TestTraceListener();
+            Log.AddListener(trace);
+            var plan = new TestPlan();
+            var parallelStep = new ParallelStep();
+            var delayStep = new ParallelStepSuggestedNextStep();
+            delayStep.RepeatCount = 10;
+            parallelStep.ChildTestSteps.Add(delayStep);
+            plan.ChildTestSteps.Add(parallelStep);
+            plan.PrintTestPlanRunSummary = true;
+            PlanRunCollectorListener planRunListener = new PlanRunCollectorListener();
+        
+            var planRun = plan.Execute(ResultSettings.Current.Concat(new IResultListener[] { planRunListener }));
```

Comment:
There is no need to append the current result settings. Overriding is fine in a unit test, unless there is a specific reason